src/main/java/teammates/storage/sqlapi/NotificationsDb.java

```diff
@@ -34,13 +37,16 @@ public static NotificationsDb inst() {
 
     /**
      * Creates a notification.
+     *
+     * <p>Preconditions:</p>
+     * * Notification fields are valid.
      */
-    public Notification createNotification(Notification notification)
-            throws InvalidParametersException, EntityAlreadyExistsException {
+    public Notification createNotification(Notification notification) throws EntityAlreadyExistsException {
         assert notification != null;
 
-        if (!notification.isValid()) {
```

Comment:
Is moving the isValid check to the logic layer better? I was thinking we could leave it here

Reply:
It was previously suggested [here](https://github.com/TEAMMATES/teammates/pull/12852#pullrequestreview-1913457548). I'm not too sure which is better myself though.